### PR TITLE
MOON-352: Add radio button moonstone component

### PR DESCRIPTION
**Pull request by jahia-carp**
https://jira.jahia.org/browse/MOON-352

### CODE REVIEW

**Review by ffffffelix — CHANGES_REQUESTED** [review]

File: src/components/Radio/Radio.scss [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+.moonstone-radio-container label[aria-readonly="true"] {
+    cursor: default;
+    opacity: 0.4;
+
+    // Prevent :hover on disabled and read-only
+    pointer-events: none;
+}
+
+.moonstone-radio-label {
+    margin-left: 8px;
```
[the remark this answers is not in the document]
Comment (jahia-carp) [reply src/components/Radio/Radio.scss]: done

File: src/components/Radio/Radio.scss [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+    color: var(--color-gray-dark);
+}
+
+.moonstone-radio-description {
+    margin-left: 24px;
```
[the remark this answers is not in the document]
Comment (jahia-carp) [reply src/components/Radio/Radio.scss]: done

File: src/components/Radio/Radio.scss [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+    border: none;
+    outline: none;
+    background: transparent;
+
+    cursor: pointer;
+
+    &:checked ~ .moonstone-radio_icon {
+        fill: var(--color-accent);
+    }
```
[the remark this answers is not in the document]
Comment (jahia-carp) [reply src/components/Radio/Radio.scss]: done

File: src/components/Radio/ControlledRadio.tsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+import clsx from 'clsx';
+import {RadioProps} from './Radio.types';
+import './Radio.scss';
+import {RadioChecked, RadioUnchecked} from '~/icons';
+import {Typography} from '~/components';
+
+export const ControlledRadio: React.FC<RadioProps> = ({className, checked = false, id, value, label, description, isDisabled, isReadOnly, ...props}) => {
+    return (
+        <div className={clsx('moonstone-radio-container flexCol', className)}>
+            <Typography aria-readonly={isReadOnly} aria-disabled={isDisabled} variant="body" htmlFor={id} weight="default" component="label">
```
[the remark this answers is not in the document]
Comment (jahia-carp) [reply src/components/Radio/ControlledRadio.tsx]: fixed, thanks :-)

File: src/components/Radio/ControlledRadio.tsx [hunk 5]
```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import clsx from 'clsx';
+import {RadioProps} from './Radio.types';
+import './Radio.scss';
+import {RadioChecked, RadioUnchecked} from '~/icons';
+import {Typography} from '~/components';
+
+export const ControlledRadio: React.FC<RadioProps> = ({className, checked = false, id, value, label, description, isDisabled, isReadOnly, ...props}) => {
+    return (
+        <div className={clsx('moonstone-radio-container flexCol', className)}>
```
[the remark this answers is not in the document]
Comment (jahia-carp) [reply src/components/Radio/ControlledRadio.tsx]: done

File: src/components/Typography/Typography.types.ts [hunk 6]
```diff
@@ -51,4 +51,8 @@ export type TypographyProps = {
      * Custom CSS style
      */
     style?: React.CSSProperties
+    /**
+     * Should be used only for label
+     */
+    htmlFor?: string;
```
[the remark this answers is not in the document]
Comment (jahia-carp) [reply src/components/Typography/Typography.types.ts]: done

File: src/components/Radio/Radio.scss [hunk 7]
```diff
@@ -0,0 +1,102 @@
+$radio-sizeDefault: 16px;
+
+.moonstone-radio-container + .moonstone-radio-container {
+    margin-top: 16px;
```
[the remark this answers is not in the document]
Comment (jahia-carp) [reply src/components/Radio/Radio.scss]: done

**Pull request merged**